### DISCUSSION TITLE
Pre-commit configuration for Zarr-Python

### DISCUSSION THREAD

**Discussion by Alt-Shivam** [zarr-developers/zarr-python#1012] [Ideas]
Hey folks,
I reviewed this issue #978 for adding pre-commit configuration.

So to start with some initial configurations,
I am starting with [this](https://github.com/ome/ome-zarr-py/blob/master/.pre-commit-config.yaml) and [this](https://tskit.dev/tskit/docs/stable/development.html#pre-commit-checks)

Although @parthxtripathi has shown interest in this issue 
But this issue is stale for the last 1 month, So I have also started working on this

will generate a PR with the initial config soon.

Till then, please post your suggestions on this Discussion.
cc @joshmoore @MSanKeys963

**Comment by sanketverma1704** [comment]
Thanks for picking this up @Alt-Shivam. Looking forward to the PR. :)

**Comment by rabernat** [comment]
:+1: to having pre-commit on Zarr. It's really helpful.

**Comment by Alt-Shivam** [comment]
So, For now, we're having flake8 in our workflow. #1015 
And I tried yamllint but got some error in the following files.
[Img1](https://files.gitter.im/625587b56da0373984945548/gs9p/image.png) [Img2](https://files.gitter.im/625587b56da0373984945548/h9Hv/image.png) [Img3](https://files.gitter.im/625587b56da0373984945548/gKGP/image.png)

So we've 3 choices now:
-Make yamllint ignore them.
-change each file according to yamllint.
-modify yamllint config

**Reply by joshmoore** [reply]
I'd suggest we start off with trying to fix.

**Reply by Alt-Shivam** [reply]
Got it.

**Comment by Alt-Shivam** [comment]
Hey folks,
While trying to make blake8 implementation in pre-commit
I noted that many .py files in Zarr-python are not formatted as per Blake8.

so should I create a PR that contains:
Blake8 and changed .py files as per blake8

Note: There are lots of files to be changed as per blake8
cc @joshmoore @rabernat @MSanKeys963

**Reply by joshmoore** [reply]
That's probably too much for the moment since it will cause many PRs to conflict. Let's get the number of open PRs down before we try that.

**Reply by Alt-Shivam** [reply]
Sure 👍🏻
